test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestProfile.java

```diff
@@ -0,0 +1,42 @@
+package io.quarkus.test.junit;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Defines a 'test profile'. Tests run under a test profile
+ * will have different configuration options to other tests.
+ *
+ */
+public interface QuarkusTestProfile {
```

Comment:
I love the idea of having all per-test configuration under one class, really simplifies things both for users and for us :)